src/runtime/finalizer.cs

```diff
@@ -97,66 +93,15 @@ public List<WeakReference> GetCollectedObjects()
 
         internal void AddFinalizedObject(IPyDisposable obj)
         {
-            if (!Enable)
-            {
-                return;
-            }
-            if (Runtime.Py_IsInitialized() == 0)
-            {
-                // XXX: Memory will leak if a PyObject finalized after Python shutdown,
-                // for avoiding that case, user should call GC.Collect manual before shutdown.
-                return;
-            }
-#if FINALIZER_CHECK
-            lock (_queueLock)
-#endif
+            if (Enable)
             {
-                _objQueue.Enqueue(obj);
-            }
-            GC.ReRegisterForFinalize(obj);
```

Comment:
Ahaha, I didn't search this fully in the past, if clr won't release the memory of object after calling the finalizer and it's needless for resurrect it, it can be remove indeed, also including the `_finalized ` member. I need some confirmation.